### PR TITLE
[VecOps] Do not enforce const-ness in Map function parameters

### PR DESCRIPTION
**Pull request by eguiraud**
Before this PR, the lambda passed to VecOps::Map could not
take arguments by non-const reference as the implementation was
adding a const qualifier to the RVec arguments.

We now instead forward RVecs from Map to MapImpl keeping their
cv qualifiers.

A test has been added.

### CODE REVIEW

**Comment by eguiraud** [comment]
@phsft-bot build

File: math/vecops/test/vecops_rvec.cxx [hunk 1]
```diff
@@ -1204,6 +1204,24 @@ TEST(VecOps, Map)
    CheckEqual(res, ref);
 }
 
+TEST(VecOps, MapOnNestedVectors)
+{
+   RVec<RVecF> vecOfVecs = {{1.f, 2.f}, {3.f}, {}};
+   RVecI sizes = {2, 1, 0};
+
```
Comment (ikabadzhov) [inline math/vecops/test/vecops_rvec.cxx:1211]: Just as an extra check maybe consider adding:
```suggestion

   auto size_by_value = [](RVecF v) { return v.size(); };
   auto result = Map(vecOfVecs, size_by_value);
   CheckEqual(result, sizes);
```

Comment (eguiraud) [reply math/vecops/test/vecops_rvec.cxx:1211]: added, thanks!

**Review by ikabadzhov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**